### DISCUSSION TITLE
Converging but very slowly

### DISCUSSION THREAD

**Discussion by Bala-1005** [idaholab/moose#17333] [Q&A Modules: Phase field] [answered]
Hello everyone,

My simulation converges quite nicely but takes a lot of time as timesteps increase. For instance, the transition from 17th to 18th time step took about 4 hours. I am not sure why since the first few timesteps only take a few minutes. 
Here is my executioner block:
[Executioner]
  type = Transient
  solve_type = 'PJFNK'
  petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type'
  petsc_options_value = 'asm      ilu          nonzero'
  l_max_its = 30
  nl_max_its = 50
  nl_abs_tol = 1.0e-10
  automatic_scaling = true
  end_time = 25
  dt = 0.008
 [./Adaptivity]
    initial_adaptivity = 3
    refine_fraction = 0.6
    coarsen_fraction = 0.1
    max_h_level = 3
  [../]
[]
[Postprocessors]
  [./dofs]
    type = NumDOFs
  [../]
  [./integral]
    type = ElementL2Error
    variable = eta
    function = ic_func_eta
  [../]
[]
Is there anything that I can do to improve the speed of the calculations? I tried removing mesh adaptivity but the solution was really bad. The problem is a phase field-mechanics problem with an initial mesh of 40*40. 
I am using a cluster system with 16 processors.
Thanks,
Bala

**Comment by GiudGiud** [comment, marked as answer]
Hi Bala

Is the mesh refinement from adaptivity happening around the 17th time step? That could explain a slowdown. You can use the [numdofs postprocessor](https://mooseframework.inl.gov/source/postprocessors/NumDOFs.html) to watch that. 

I think the output log may be useful here to evaluate how the system is converging. Could you please attach it here

Guillaume

**Reply by Bala-1005** [reply]
Hi Guillaume,

Thank you for your suggestion.
Mesh refinement is happening at every time step. I think I may have miswritten when I said 16th and 17th timestep. There are about 30 timesteps in between two timesteps when output is written. So it takes about 3 hours for 30 timesteps is what I meant. 
I have attached the output file with the performance log. 

Thanks, 
Bala
[el2b.txt](https://github.com/idaholab/moose/files/6152279/el2b.txt)

**Reply by GiudGiud** [reply]
From your log it seems that a lot of time steps are failing because 10 non linear iterations (what I m seeing in the log, the block you pasted says 50) is not enough to reach the target residual. You should consider:
- using a relative non linear tolerance criterion instead of an absolute
- allowing more than 10 non linear iterations so the time steps fail less often

You could also increase the time step size if that is wise. 

Have you tried the Newton solver? It might work better for your application.

In the log I see 28 processes. If your machine has 16 cores it's likely 16 processes will give the best performance.

**Reply by Bala-1005** [reply]
I m not sure what happened with the log file. Let me try all these suggestions that you have given me and I will update on the status of the problem. Thank you for all the help.

**Reply by Bala-1005** [reply]
Allowing more than 10 nonlinear iterations seem to help the convergence issue to a certain extent. The problem that I am running is a solidification problem. When two solid parts meet in the simulation the solution stops converging. I have referenced this problem in another related issue #16922 as well. I am attaching the output log file along with the debug file. It is kind of hard to figure out where the issue is. Any help would be greatly appreciated. 
One of the performance output from the file is as below

 Heaviest Sections:
 ---------------------------------------------------------------------------------------
 |                  Section                 | Calls |   Self(s)  |    Avg.    |    %   |
 ---------------------------------------------------------------------------------------
 | NonlinearSystemBase::computeJacobianTags |  5272 |    144.864 |      0.027 |  32.74 |
 | FEProblem::solve                         |   210 |     82.631 |      0.393 |  18.67 |
 | DisplacedProblem::updateMesh             | 11164 |     46.140 |      0.004 |  10.43 |
 | FEProblem::checkExceptionAndStopSolve    | 72902 |     43.131 |      0.001 |   9.75 |
 | NonlinearSystemBase::Kernels             |  5714 |     35.089 |      0.006 |   7.93 |
 | AuxiliarySystem::computeElementalVars    |  5924 |     31.754 |      0.005 |   7.18 |
 | FEProblem::meshChangedHelper             |   177 |     30.736 |      0.174 |   6.95 |
 ---------------------------------------------------------------------------------------
 


Thanks,
Bala
[el2b_KHS_error.txt](https://github.com/idaholab/moose/files/6167106/el2b_KHS_error.txt)
[el2b_KHS_output.txt](https://github.com/idaholab/moose/files/6167107/el2b_KHS_output.txt)

**Reply by GiudGiud** [reply]
continued in #16922
